infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -113,28 +191,105 @@ public long getLastSnapSeqNum() {
         return query(null, LogReplicationMetadataType.LAST_SNAPSHOT_SEQ_NUM);
     }
 
+    public long getLastSnapAppliedSeqNum() {
+        return query(null, LogReplicationMetadataType.LAST_SNAPSHOT_APPLIED_SEQ_NUM);
+    }
+
     public long getLastProcessedLogTimestamp() {
         return query(null, LogReplicationMetadataType.LAST_LOG_PROCESSED);
     }
 
-    public void appendUpdate(TxBuilder txBuilder, LogReplicationMetadataType key, long val) {
+    /**
+     * Given a specific timestamp, get the most recent full snapshot sync's timestamp that has complete the
+     * transferred phase.
+     * If the timestamp is null, get the most recent value.
+     * @param ts
+     * @return
+     */
+    public long getLastSnapTransferDoneTimestamp(CorfuStoreMetadata.Timestamp ts) {
```

Comment:
Do we really need explicitly query for individual fields within the metadata, wouldn't it be simpler to just return the actual metadata itself?

Reply:
I am planning to revisit the metadata for the replication status in another PR to address these issues.
Like you suggest before, group them to one data structure as value and with key as "currentLogReplicationData". This reduce the number of access of DB too.